### ISSUE TITLE
train_mnist.py with 4 GPUs is slower than itself with 1GPU

### ISSUE THREAD

**Issue by DamonDeng** [dmlc/mxnet#4286]
For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System:
Ubuntu

Compiler:
g++

Package used (Python/R/Scala/Julia):
Python

MXNet version:
0.7.0

Or if installed from source:
Yes

MXNet commit hash (`git rev-parse HEAD`):
1036277b96baf96418385a5f722d989f18a040bf

If you are using python package, please provide

Python version and distribution:
2.7

If you are using R package, please provide

R `sessionInfo()`:

## Error Message:
Please paste the full error message, including stack trace.
No error message

## Minimum reproducible example
if you are using your own code, please provide a short script that reproduces the error.

Run the following two command to run the train_mnist.py with 1 GPU and 4 GPU, the one with 4 GPU is slower than the one with 1 GPU:

###Command:

`python ../mxnet_install/mxnet/example/image-classification/train_mnist.py --gpus 0`

`python ../mxnet_install/mxnet/example/image-classification/train_mnist.py --gpus 0,1,2,3`

### out put with 1 GPU:

python ../mxnet_install/mxnet/example/image-classification/train_mnist.py --gpus 0
INFO:root:start with arguments Namespace(batch_size=64, disp_batches=100, gpus='0', kv_store='device', load_epoch=None, lr=0.1, lr_factor=0.1, lr_step_epochs='1
0', model_prefix=None, mom=0.9, network='mlp', num_classes=10, num_epochs=20, num_examples=60000, num_layers=None, optimizer='sgd', test_io=0, top_k=0, wd=0.000
1)
INFO:root:Start training with [gpu(0)]
INFO:root:Epoch[0] Batch [100]  Speed: 36736.40 samples/sec     Train-accuracy=0.785937
INFO:root:Epoch[0] Batch [200]  Speed: 36514.88 samples/sec     Train-accuracy=0.893906
INFO:root:Epoch[0] Batch [300]  Speed: 36543.26 samples/sec     Train-accuracy=0.916562
INFO:root:Epoch[0] Batch [400]  Speed: 36632.78 samples/sec     Train-accuracy=0.935625
INFO:root:Epoch[0] Batch [500]  Speed: 36715.75 samples/sec     Train-accuracy=0.936875
INFO:root:Epoch[0] Batch [600]  Speed: 37090.91 samples/sec     Train-accuracy=0.939688
INFO:root:Epoch[0] Batch [700]  Speed: 37159.15 samples/sec     Train-accuracy=0.947812
INFO:root:Epoch[0] Batch [800]  Speed: 37115.01 samples/sec     Train-accuracy=0.947656
INFO:root:Epoch[0] Batch [900]  Speed: 37071.55 samples/sec     Train-accuracy=0.950625
INFO:root:Epoch[0] Resetting Data Iterator
INFO:root:Epoch[0] Time cost=1.945
INFO:root:Epoch[0] Validation-accuracy=0.953324

### out put of 4 GPUs

python ../mxnet_install/mxnet/example/image-classification/train_mnist.py --gpus 0,1,2,3
INFO:root:start with arguments Namespace(batch_size=64, disp_batches=100, gpus='0,1,2,3', kv_store='device', load_epoch=None, lr=0.1, lr_factor=0.1, lr_step_epo
chs='10', model_prefix=None, mom=0.9, network='mlp', num_classes=10, num_epochs=20, num_examples=60000, num_layers=None, optimizer='sgd', test_io=0, top_k=0, wd
=0.0001)
INFO:root:Start training with [gpu(0), gpu(1), gpu(2), gpu(3)]
INFO:root:Epoch[0] Batch [100]  Speed: 17725.99 samples/sec     Train-accuracy=0.782813
INFO:root:Epoch[0] Batch [200]  Speed: 17591.45 samples/sec     Train-accuracy=0.907500
INFO:root:Epoch[0] Batch [300]  Speed: 17856.25 samples/sec     Train-accuracy=0.930312
INFO:root:Epoch[0] Batch [400]  Speed: 18414.99 samples/sec     Train-accuracy=0.934219
INFO:root:Epoch[0] Batch [500]  Speed: 17935.61 samples/sec     Train-accuracy=0.935469
INFO:root:Epoch[0] Batch [600]  Speed: 18190.92 samples/sec     Train-accuracy=0.942031
INFO:root:Epoch[0] Batch [700]  Speed: 18301.91 samples/sec     Train-accuracy=0.944375
INFO:root:Epoch[0] Batch [800]  Speed: 17891.83 samples/sec     Train-accuracy=0.944688
INFO:root:Epoch[0] Batch [900]  Speed: 17640.18 samples/sec     Train-accuracy=0.946562
INFO:root:Epoch[0] Resetting Data Iterator
INFO:root:Epoch[0] Time cost=3.803
INFO:root:Epoch[0] Validation-accuracy=0.954618

## Steps to reproduce
or if you are running standard examples, please provide the commands you have run that lead to the error.

1. Launch AWS G2.8x large
2. Install MXNet
3. Run the training with 1 GPU and 4 GPU

## What have you tried to solve it?

1. I tried to set  `export MXNET_ENABLE_GPU_P2P=0` then re-run the training, but doesn't help
2. I tried to search in the issue list of this repository, didn't find duplicate case.
3.

**Comment by pineking** [comment]
batch_size is set to 64  in 1 GPU case in your experiment, so set it to 64*4=256 in 4 GPUs case
btw: try to run some huge computation cost experiments, such as cifar10, imagenet, the mnist database is too simple, the speed of 1 GPU shown in the log is fast enough.

**Comment by DamonDeng** [comment]
Many thanks for your response, I tried to set the --batch-size parameter to 256, the one with 1 GPU still ran faster than the one with 4 GPU. It make sense to me that the mnist data set is too simple, I tried train_cifa10.py, got about 250 samples per second with 1 GPU , and got about 600 samples per second with 4 GPU.

Will close this issue
